lib/puppet/pops/types/type_calculator.rb

```diff
@@ -1192,22 +1201,34 @@ def tuple_entry_at(tuple_t, from, to, index)
   # @api private
   #
   def assignable_PStructType(t, t2)
-    return true if t == t2 || t.elements.empty? && (t2.is_a?(Types::PHashType))
-    h = t.hashed_elements
+    els = t.elements
+    return true if t == t2 || els.empty? && (t2.is_a?(Types::PHashType))
```

Comment:
This is odd - accepting any Hash if empty is fine, since if the t Struct has no elements it means any struct, and thus any hashtype, but that is also true of any struct type.